Lib/lib2to3/tests/test_parser.py

```diff
@@ -212,12 +215,33 @@ def test_multiline_str_literals(self):
         self.validate(s)
 
 
+
 def diff(fn, result, encoding):
-    "A diff the result and original file content independent of OS."
+    """Diff the result and original file content independent of OS. This 
+    implementation is in Jython only. CPython 2.x calls out to the shell diff
+    command. CPython 3.x str equality behaviour is simpler and should allow
+    this test file to be replaced wholesale in Jython 3.x."""
     r = iter(result.encode(encoding).splitlines(True))
-    with open(fn, "rb") as f:
+    lineNumber = 0
+    with open(fn, "r") as f:
         for line in f:
+            lineNumber += 1
             rline = next(r)
             if rline != line:
-                return True
+                return "original line {}:\n{}\n differs from:\n{}\n{}". \
+                       format(lineNumber,line,rline,diffLine(line,rline) )
```

Comment:
Comma-space, here and elsewhere. The CPython test suite is supposed to be PEP-8. We've been a bit lax, but some things are easy. (Perhaps I should use a linter.)

Reply:
Will fix

Reply:
Fixed in latest for the few jython-specific lines. Rest I kept as character identical to merge from CPython